include/Parser.h

```diff
@@ -20,6 +20,8 @@ class Preprocessor;
 /// Implements a full syntax parser for SystemVerilog.
 class Parser : ParserBase {
 public:
+    static constexpr size_t MaxDepth=100;
```

Comment:
Could you add a TODO that we should pull this value from an Options system at some point (which doesn't currently exist)? That way things like this can be configured from the command line.